### PR TITLE
Cargo: bump llvm-ir

### PR DESCRIPTION
**Pull request by woodruffw**
Signed-off-by: William Woodruff <william@trailofbits.com>

### CODE REVIEW

**Comment by woodruffw** [comment]
I went ahead and used this same branch to clean the README up a bit, but I can break that out if desired.

File: README.md [hunk 1]
```diff
@@ -9,51 +18,133 @@ Badness  Function
...
+Functions with high badness have a lot of opportunities to be memory unsafe.
 
 ## Installation
 
-Make sure that you have the following requirements:
+Siderophile is [available via crates.io](https://crates.io/crates/siderophile),
+and can be installed with `cargo`:
+
+```console
+$ cargo install siderophile
```
Comment (smoelius) [inline README.md:33]:
```suggestion
cargo install siderophile
```
`$` don't play nicely with GitHub's click-to-copy feature. So I recommend not including them.

File: README.md [hunk 2]
```diff
@@ -9,51 +18,133 @@ Badness  Function
...
+error: could not compile `llvm-sys` due to previous error
+```
+
+This indicates that the build was unable to automatically find a copy of LLVM to link against.
+
+You can fix it by setting the `LLVM_SYS_140_PREFIX`. For example, for macOS with LLVM via
+Homebrew, you might do:
+
+```console
+$ LLVM_SYS_140_PREFIX=$(brew --prefix)/opt/llvm/ cargo install siderophile
```
Comment (smoelius) [inline README.md:67]:
```suggestion
LLVM_SYS_140_PREFIX=$(brew --prefix)/opt/llvm/ cargo install siderophile
```

File: README.md [hunk 3]
```diff
@@ -9,51 +18,133 @@ Badness  Function
...
 
-- `rustup` and `cargo` must be installed and in your `PATH`
-- LLVM 13 is required. Older versions may work (see https://crates.io/crates/llvm-ir) but require the `llvm-ir` package's features change in `Cargo.toml` before compiling Siderophile.
+### Building and installing from source
 
-Then, run `cargo build --release`, and you'll have a Siderophile binary :)
+Alternatively, if you'd like to build from source:
+
+```console
+$ git clone https://github.com/trailofbits/siderophile && cd siderophile
```
Comment (smoelius) [inline README.md:75]:
```suggestion
git clone https://github.com/trailofbits/siderophile && cd siderophile
```

File: README.md [hunk 4]
```diff
@@ -9,51 +18,133 @@ Badness  Function
...
+
+# TIP: include --release for a release build
+$ cargo build
```
Comment (smoelius) [inline README.md:78]:
```suggestion
cargo build
```

File: README.md [hunk 5]
```diff
@@ -9,51 +18,133 @@ Badness  Function
...
+
+# optionally: install the built binary to cargo's default bin path
+$ cargo install --path .
```
Comment (smoelius) [inline README.md:81]:
```suggestion
cargo install --path .
```

File: README.md [hunk 6]
```diff
@@ -9,51 +18,133 @@ Badness  Function
...
+we can match with the source code.
+
+Next, Siderophile finds all the sources of the current crate, finds every
+Rust file in the sources, and parses each file individually using the `syn`
+crate. Each file is recursively combed through for unsafety occurring in
+functions, trait declarations, trait implementations, and submodules.
+Siderophile will output the path of these objects, along with an indication
+of what type of syntactic block they were found in. The list received from this
+step contains every unsafe block in every dependency of the crate, regardless
+of whether it's used. To narrow this down, we need to compare Siderophile's
```
Comment (smoelius) [inline README.md:110]:
```suggestion
of whether it's used. To narrow this down, Siderophile needs to compare its
```

File: README.md [hunk 7]
```diff
@@ -9,51 +18,133 @@ Badness  Function
...
+list to nodes in the callgraph of the crate.
+
+Using the callgraph produced in the first step, we check which elements from the
```
Comment (smoelius) [inline README.md:113]:
```suggestion
Using the callgraph produced in the first step, Siderophile checks which elements from the
```

File: README.md [hunk 8]
```diff
@@ -9,51 +18,133 @@ Badness  Function
...
+Siderophile output are actually executed from the crate in question. This
```
Comment (smoelius) [inline README.md:114]:
```suggestion
output are actually executed from the crate in question. This
```
Just to eliminate the repeated "Siderophile."

File: README.md [hunk 9]
```diff
@@ -9,51 +18,133 @@ Badness  Function
...
+increment their badness by one, thus indicating that they use unsafety at some
+point in their execution. At the end of this process, all the nodes with nonzero
+badness are printed out, sorted in descending order by badness.
 
 ## Limitations
 
-Siderophile is _not_ guaranteed to catch all the unsafety in a crate's deps. Since things are only tagged at a source-level, we do not have the ability to inspect macros or resolve dynamically dispatched methods. Accordingly, this tool should not be used to "prove" that a crate uses no unsafety.
+Siderophile is _not_ guaranteed to catch all the unsafety in a crate's deps.
+
+Since things are only tagged at a source-level, we do not have the ability to
```
Comment (smoelius) [inline README.md:130]:
```suggestion
Since things are only tagged at a source-level, Sierophile does not have the ability to
```

**Review by smoelius — APPROVED** [review]
The changes to the code itself look fine to me. Thanks for doing this.

**Pull request merged**